3_visualize/src/drought_learner_viz_components.R

```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+    filter(start_year == end_year)
+  p2_droughts_site <- bind_rows(p2_droughts_site_wrapYr,
+                                p2_droughts_site_noWrapYr)
+  
+  
+  blank_plot_year <- ggplot(data = p2_streamflow_learner_viz_df, aes(y = value, x = dt))+
+    ylab("Year\n(sort of)")+
+    xlab(NULL)+
+    scale_x_date(labels = date_format("%b"), 
+                 date_breaks  ="1 month",
+                 limits = c(as.Date("01/01/1963",'%d/%m/%Y'), as.Date("31/12/1963",'%d/%m/%Y')))+
+    theme_tufte(base_family = "sans", base_size = 16)+
+    theme(axis.line = element_line(color = 'black'),
+          axis.text = element_text(size = 6),
+          axis.title = element_text(size = 8),
+          panel.background = element_blank())
+  
+  main <- blank_plot_year +
+    # Fixed threshold drought durations
+    annotate("rect", # fixed threshold
+             xmin = (p2_droughts_site$start_fakeYr),
+             xmax = (p2_droughts_site$end_fakeYr),
+             ymin = p2_droughts_site$start_year-1, ymax = p2_droughts_site$start_year,
+             fill = "#da7d81", alpha = 0.9) +
+    # Variable threshold drought durations
+    annotate("rect", # variable threshold
+             xmin = (p2_droughts_jd$start_fakeYr),
+             xmax = (p2_droughts_jd$end_fakeYr),
+             ymin = p2_droughts_jd$start_year-70, ymax = p2_droughts_jd$start_year-69,
+             fill = "#e6af84", alpha = 0.9)+
+    theme(axis.title.y = element_text(color="transparent"),
+          axis.text.y = element_text(color = "transparent"),
+          axis.ticks.y = element_line(color = "transparent"))
+  
+  
+  # Bottom bars
+  bottom_bars_year <- blank_plot_year + 
+    annotate("rect", # fixed threshold
+             xmin = (p2_droughts_learner_viz_df$start[p2_droughts_learner_viz_df$method == "fixed"]),
+             xmax = (p2_droughts_learner_viz_df$end[p2_droughts_learner_viz_df$method == "fixed"]),
+             ymin = 52, ymax = 100,
+             fill = "#da7d81", alpha = 0.8)+
+    annotate("rect", # variable threshold
+             xmin = (p2_droughts_learner_viz_df$start[p2_droughts_learner_viz_df$method == "variable"]),
+             xmax = (p2_droughts_learner_viz_df$end[p2_droughts_learner_viz_df$method == "variable"]),
+             ymin = 0, ymax = 48,
+             fill = "#e6af84", alpha = 0.8)+
+    theme(axis.title = element_text(color = "transparent"),
+          axis.text = element_text(color = "transparent"),
+          axis.line = element_line(color = "transparent"),
+          axis.ticks = element_line(color = "transparent"))
+  
+  ggdraw(ylim = c(0,1), 
+         xlim = c(0,1)) +
+    # fill in the background
+    draw_grob(canvas,
+              x = 0.25, y = 1,
+              height = 12, width = 16,
+              hjust = 0, vjust = 1) +
+    # Main plot
+    draw_plot(main,
+              x = 0,
+              y = 0.1,
+              height = 0.9,
+              width = 1)+
+    # Inset map
```

Comment:
RC: I think the inset map should either be move or dropped here, since it's blocking some of the data